embedded/sql/stmt.go

```diff
@@ -2156,17 +2156,21 @@ func (stmt *SelectStmt) Resolve(tx *SQLTx, params map[string]interface{}, _ *Sca
 	}
 
 	if stmt.joins != nil {
-		rowReader, err = newJointRowReader(rowReader, stmt.joins)
+		jointRowReader, err := newJointRowReader(rowReader, stmt.joins)
```

Comment:
I'm wondering if we cannot defer the close of rowReader similarly as suggested in previous file...